pyrovision/inference/inference.cfg

```diff
@@ -0,0 +1,13 @@
+[MODEL]
+backbone = rexnet1_0x
+num_classes = 1
+use_adaptive_concat_pool2d_head = True
+nb_features = 1280
+cut = -2
+url = https://github.com/pyronear/pyro-vision/releases/download/v0.1.0/rexnet1_0x_acp_2e017f83.pth
+device = cuda
+
+[TRANSFORMS]
+image_size = 448
+use_center_crop = True
+
```

Comment:
We cannot afford to add a config here for each model that we will train. But we could pass these params to the function instantitating the model, and if `pretrained=True`, download the checkpoint from the URL & load it

Reply:
Of course you can't have one configuration file per model. The goal here is to do it only for our reference model. The one we use for all our predictions. This reference model will then be used wherever we need it on the raspberry pi, on the web app ...
What do you think about it ?

Reply:
Well my point is that this information could be stored as model attributes and set by constructor arguments. I'm not saying we don't need any of these. If I were to check each element here:
- backbone info : can be obtained by checking the class name
- num_classes : easily stored in the model or by checking the size of the last FC layer
- use_adaptive : as a constructor argument it's important but doesn't need to be stored afterwards (ie. we put this value as default in the function which will work with the pretrained model we have made)
- nb_features : is it to build the FC? if so, it can either be read as the last channel number in conv layers if there is a Global pooling, or dynamically by forwarding a mock input (and it is important because it depends then on the image size)
- cut : constructor /input arg is relevant, not sure it is useful as attribute
- url : should be stored in a static dict in the module
- device: i'd argue it should not be set. The model state_dict needs to be saved with CPU tensor by safety, but then the user will choose on which type of device she/he will use the model.
- transforms: input image size is relevant, but center crop isn't. Either you mean it as augmentation and it's for training, not inference, or we lose information in the image --> in which case this needs to be changed on inference: we don't want the model to say there is no fire if it's just outside the center crop while it was clearly in the input image and the model could have detected it.

All relevant information related to the model should be "linked" to it (as attributes). If we put them in a different folder/file, and forget to change them after updating a model, this will be troublesome :sweat_smile:  For me there are only two types of valuable information for inference purposes:
- how we build the model (constructor arguments, potentially used as attributes afterwards)
- how we prepare model's input / understand model outputs (image size, resizing/cropping scheme, normalization parameters, number of output classes)

The rest doesn't seem relevant for inference in my opinion :)
My apologies for the long reply haha